src/features/controls.ts

```diff
@@ -361,11 +371,19 @@ export const ControlsMixin = (ModelViewerElement:
           this[$promptElement].classList.remove('visible');
         }
 
+        [$onBegin]() {
+          this.dispatchEvent(new CustomEvent('interact-start'));
+        }
+
         [$onChange]() {
```

Comment:
This handler is actually invoked every time that the camera position changes due to the `SmoothControls` instance taking action on it. If you echoed this event (let's say we call it `camera-change` or something), then you would automatically get support for keyboard, mouse, touch etc.

Here is where we dispatch the Three.js-style event in the `SmoothControls` implementation:

https://github.com/GoogleWebComponents/model-viewer/blob/dd4cd75d8b41bfd68b715e781a883df2050105e9/src/three-components/SmoothControls.ts#L408-L412

And that event is handled here in the `$onChange` handler. The downside of this approach is that it is very noisy (literally every incremental change, no matter how minute, causes the handler to be invoked). But, that might be fine in practice. It wouldn't be any more noisy than what the browser already does for `touchmove` or `mousemove` events.

WDYT?

Reply:
But you'd need to toggle the ability to auto rotate somehow. Of course every time change is fired we could set the timer to start to count down auto rotate.

One reason I wanted to stay away from `camera-change` is that I think that event should return some sort of data. I guess the `camera-orbit` information would be handy to return.

Reply:
> every time change is fired we could set the counter/timer to start to count down auto rotate.

Yes, this is roughly what I had in mind.

> One reason I wanted to stay away from camera-change is that I think that event should return some sort of data. I guess the camera-orbit information would be handy to return.

This is a good point. For the first pass, I would recommend that we dispatch the event and not add any additional information. The reason why is because the user can technically get the information using the public API we added in #452 . So, if they needed the camera orbit details, they can do something like this:

```javascript
modelViewer.addEventListener('camera-change', (event) => {
  const orbit = modelViewer.getCameraOrbit();
  // ... or ...
  const orbit = event.target.getCameraOrbit();
});
```

This would at least save us from serializing the spherical on every event dispatch.

Reply:
Also, now that I think about it, `camera-change` isn't a great name because it could also mean that the active camera changed to something else (we don't support this now, but we might in the future).

Something like `camera-position-change` or `camera-orbit-change` might be better.

WDYT?

Reply:
I like this thinking. 

I'll think about what the event should be named. It would be pretty nice to start an issue re what events `<model-viewer>` should and shouldn't dispatch.

To me this is really important to allow for customization and generally interoperability with existing systems.